### PR TITLE
Better replace for actors

### PR DESCRIPTION
**Pull request by PierreAndreis**
#23

### CODE REVIEW

**Comment by PierreAndreis** [comment]
Somehow it passes the tests once I moved the function from Utils to resources

**Comment by PierreAndreis** [comment]
@seripap We need support for Lorelai (released yesterday), can we get this PR pushed?

**Comment by seripap** [comment]
Sorry about the timing on this, been a bit busy on this side. Will take a look now.

File: src/models/participant.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import BaseModel from './';
 import actors from './resources/actors';
+import decodeActorName from './resources/actors_decode';
```
Comment (seripap) [inline src/models/participant.js]: I don't think this is needed as a separate utility file as its only function is to replace actor names. We can keep it in the scope of the `actor` getter.

File: src/models/resources/actors_decode.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export function decodeActorName(token) {
+  // Returns the hero name, given a token.
+  //
+  // For "*Hero009*", etc., returns "Krul", using heroTokensBad look-up table.
+
+  let match = /^\*(.*)\*$/.exec(token);
+  if (match) return (heroTokensBad[token] || match[1]);
+
+  // If somehow it doesn't match, let's just return the token
+  return token;
```
Comment (seripap) [inline src/models/resources/actors_decode.js]: This is fine, but keeping with the standard set with heroTokensBad, we'd probably want to remove all instances of the asterisk. 

```js
return token.replace(/\*/g, '')
```

File: src/models/participant.js [hunk 3]
```diff
@@ -41,8 +42,7 @@ export default class Participant extends BaseModel {
   }
 
   get actor() {
-    const normalizedActor = actors.find(actor => actor.serverName === this.raw.attributes.actor);
-    return normalizedActor ? normalizedActor.name : this.raw.attributes.actor;
+    return decodeActorName(this.raw.attributes.actor);
```
Comment (seripap) [inline src/models/participant.js]: Move all over actors decode stuff in here.

File: src/models/resources/actors_decode.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  "*Hero016*"   : "Rona",
+};
+
+export function decodeActorName(token) {
+  // Returns the hero name, given a token.
+  //
+  // For "*Hero009*", etc., returns "Krul", using heroTokensBad look-up table.
+
+  let match = /^\*(.*)\*$/.exec(token);
+  if (match) return (heroTokensBad[token] || match[1]);
```
Comment (seripap) [inline src/models/resources/actors_decode.js]: Given the current code style, can you wrap this with braces?

**Review by seripap — CHANGES_REQUESTED** [review]
Some minor adjustments. Will merge tonight once updated. Thanks for your work :+1:

**Comment by seripap** [comment]
One other thing, could you also remove the file `models/resources/actors.js` since it's not being used anymore after this PR?

File: src/models/resources/actors_decode.js (hunk 4 above)
Comment (PierreAndreis) [reply src/models/resources/actors_decode.js]: Can you give me an example where this code style is in use?

**Review by seripap — APPROVED** [review]

**Pull request merged**